pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolHandler.java

```diff
@@ -0,0 +1,590 @@
+        public TrailerOnlyBuilder httpStatus(@Nullable final Status httpStatus) {
+            this.httpStatus = httpStatus;
+            return this;
+        }
+
+        /**
+         * Send the headers back to the client
+         * @param httpHeaders The normal HTTP headers (also grpc headers)
+         * @param http2Headers The HTTP2 pseudo-headers
+         */
+        @Override
+        protected void send(
+                @NonNull final WritableHeaders<?> httpHeaders,
+                @NonNull final Http2Headers http2Headers) {
+            http2Headers.status(httpStatus);
+            httpHeaders.contentType(requireNonNull(contentType));
+            super.send(httpHeaders, http2Headers);
+        }
+    }
+
+    /**
+     * The implementation of {@link Flow.Subscriber} used to send messages to the client.
+     * It receives bytes from the handlers to send to the client.
+     */
+    private final class SendToClientSubscriber implements Flow.Subscriber<Bytes> {
+        @Override
+        public void onSubscribe(@NonNull final Flow.Subscription subscription) {
+            // FUTURE: Add support for flow control
+            subscription.request(Long.MAX_VALUE);
+        }
+
+        @Override
+        public void onNext(@NonNull final Bytes response) {
+            try {
+                final int length = (int) response.length();
+                final var bufferData = BufferData.create(5 + length);
+                bufferData.write(0); // 0 means no compression
+                bufferData.writeUnsignedInt32(length);
+                bufferData.write(response.toByteArray());
+                final var header = Http2FrameHeader.create(bufferData.available(),
+                        Http2FrameTypes.DATA,
+                        Http2Flag.DataFlags.create(0),
+                        streamId);
+
+                streamWriter.writeData(new Http2FrameData(header, bufferData), flowControl.outbound());
+            } catch (final Exception e) {
+                LOGGER.log(ERROR, "Failed to respond to grpc request: " + route.method(), e);
+            }
+        }
+
+        @Override
+        public void onError(@NonNull final Throwable throwable) {
+            if (throwable instanceof GrpcException grpcException) {
+                new TrailerBuilder().grpcStatus(grpcException.status())
+                        .statusMessage(grpcException.getMessage())
+                        .send();
+            } else {
+                LOGGER.log(ERROR, "Failed to send response", throwable);
+                new TrailerBuilder().grpcStatus(GrpcStatus.INTERNAL)
+                        .send();
+            }
+            close();
+        }
+
+        @Override
+        public void onComplete() {
+            new TrailerBuilder().send();
+            close();
+        }
+    }
+
+    /**
+     * Simple implementation of the {@link ServiceInterface.RequestOptions} interface.
+     */
+    private record Options(
+            String authority,
```

Comment:
Should this be `Optional<String>` to match the interface being implemented?

Reply:
Fixed